app/src/main/java/com/example/newsapp/presentation/home/HomeViewModel.kt

```diff
@@ -16,6 +16,8 @@ import kotlinx.coroutines.launch
 class HomeViewModel(
     private val newsUseCase: GetNewsUseCase,
     private val profileUseCase: GetProfileUseCase,
+    private val increment: (() -> Unit)? = null,
```

Comment:
I don't know this is better or not, because expose to the view model constructor. you can find another way I think

Reply:
I have tried directly injection EspressoIdlingResource.increment() and decrement() and it works for UI testing, but when I run unit test, it fails mas.. it says it's not mockable.. so I did constructor injection, so that in unit test I can simply put an empty function in the viewmodel.. do u have an idea mas?

Reply:
better empty object {} instead null,